posix.mak

```diff
@@ -652,17 +652,35 @@ test: $(ASSERT_WRITELN_BIN)_test all
 
 ################################################################################
 # Changelog generation
+#
+# pending:		$(LATEST)..upstream/master		(all changes of the next _major_ release)
+# pre:			$(LATEST)..upstream/stable 		(all changes of the next release)
+# release:		$(PREVIOUS_LATEST)..$(LATEST)   (used by the release maintainer after updating $(LATEST))
 ################################################################################
 
 changelog/${NEXT_VERSION}_pending.dd: | ${STABLE_DMD} ../tools ../installer
 	$(STABLE_RDMD) $(TOOLS_DIR)/changed.d $(CHANGELOG_VERSION_MASTER) -o $@ \
 	--version "${NEXT_VERSION} (pending)" --date "Pending" --nightly
 
+pending_changelog: changelog/${NEXT_VERSION}_pending.dd html
+	@echo "Please open file:///$(shell pwd)/web/changelog/${NEXT_VERSION}_pending.html in your browser"
+
 changelog/${NEXT_VERSION}_pre.dd: | ${STABLE_DMD} ../tools ../installer
 	$(STABLE_RDMD) $(TOOLS_DIR)/changed.d $(CHANGELOG_VERSION_STABLE) -o $@ \
-		--version "${NEXT_VERSION}"
+		--version "${NEXT_VERSION} (beta)"  --date "To be released"
 
-pending_changelog: changelog/${NEXT_VERSION}_pending.dd html
-	@echo "Please open file:///$(shell pwd)/web/changelog/${NEXT_VERSION}_pending.html in your browser"
+pre_changelog: changelog/${NEXT_VERSION}_pre.dd html
+	@echo "Please open file:///$(shell pwd)/web/changelog/${NEXT_VERSION}_pre.html in your browser"
+
+# --no-text is used for point-releases like 2.074.1 to exclude plain-text files in changelog/* folders
+#  which are inteded for major releases only.
+changelog/${LATEST}.dd: | ${STABLE_DMD} ../tools ../installer
+	previous_latest=$$(git tag | grep -vE "(b|rc)" | tail -n2 | head -n1) && \
+	changelog_flags=$$(LATEST=$(LATEST) && [ "$${LATEST: -1}" -gt 0 ] && echo "--no-text") && \
+	$(STABLE_RDMD) $(TOOLS_DIR)/changed.d "$$(echo $${previous_latest}..v$(LATEST))" $$changelog_flags \
+		-o "changelog/$(LATEST).dd" --version "$(LATEST)" --date "$$(date +'%B %d, %Y')"
+
+release_changelog: changelog/${LATEST}.dd html $(DOC_OUTPUT_DIR)/changelog/$(LATEST).html
```

Comment:
@MartinNowak `release_changelog` should be a drop-in replacement for whatever you are currently using. It's a bit different:

```diff
diff --git a/changelog/2.074.1.dd b/changelog/2.074.1.dd
index f1e38b17..f497c8ce 100644
--- a/changelog/2.074.1.dd
+++ b/changelog/2.074.1.dd
@@ -2,17 +2,13 @@ Ddoc
 
 $(CHANGELOG_NAV_LAST 2.074.0)
 
-$(VERSION May 30, 2017, =================================================,
+$(VERSION July 15, 2017, =================================================,
 
-$(BR)$(BIG $(RELATIVE_LINK2 bugfix-list, List of all bug fixes and enhancements in D $(VER).))
-
-$(HR)
-
-$(BR)$(BIG $(LNAME2 bugfix-list, List of all bug fixes and enhancements in D $(VER):))
+$(BR)$(BIG List of all bug fixes and enhancements in D $(VER).)
 
 $(BUGSTITLE DMD Compiler regressions,
 
-$(LI $(BUGZILLA 17407): [REG] __traits$(LPAREN)compiles$(RPAREN) triggers assertion failure)
+$(LI $(BUGZILLA 17407): [REG2.067] __traits$(LPAREN)compiles$(RPAREN) triggers assertion failure)
 )
 $(BUGSTITLE DMD Compiler bugs,
 
@@ -20,6 +16,7 @@ $(LI $(BUGZILLA 17289): With Xcode 8.3 linker, warnings of "pointer not aligned"
 )
 $(BUGSTITLE Phobos regressions,
 
+$(LI $(BUGZILLA 17264): [REG2.073] uniq fails with const elements)
 $(LI $(BUGZILLA 17330): [REG 2.072] BigInt's constructor used to be pure)
 $(LI $(BUGZILLA 17340): [REG 2.074.0] isNumeric!bool should not be true)
 )
```